app/presenters/course_for_menu_presenter.rb

```diff
@@ -42,7 +42,7 @@ def to_h
       id: course.id,
       isFavorited: course.favorite_for_user?(@user) && @user.account.feature_enabled?(:unfavorite_course_from_dashboard),
       image: course.feature_enabled?(:course_card_images) ? course.image : nil,
-      position: @user.dashboard_positions[course.asset_string] || nil,
+      position: @user.dashboard_positions[course.asset_string].to_i || nil,
```

Comment:
I don't think this is quite what we want, because `nil.to_i == 0` in ruby, so if the user doesn't have a position set for a course, this will make it 0, instead of falling back to the `|| nil` case.

I think you'll have to do something like:
```
if @user.dashboard_positions[course.asset_string].present?
  @user.dashboard_positions[course.asset_string].to_i
else
  nil
end
```
or 
```
pos = @user.dashboard_positions[course.asset_string]
pos.present? ? pos.to_i : nil
```

Reply:
Thanks for the suggestion. I added the code that you suggested and updated the test.

Jonathans-MacBook-Pro:canvas-lms jonathannovecio$ docker-compose run --rm web bundle exec rspec spec/helpers/dashboard_helper_spec.rb
Starting canvas-lms_redis_1 ... done
Starting canvas-lms_postgres_1 ... done
/home/docker/.gem/ruby/2.4.0/gems/soap4r-ruby1.9-2.0.5/lib/soap/mapping/encodedregistry.rb:150: warning: constant ::Fixnum is deprecated
/home/docker/.gem/ruby/2.4.0/gems/soap4r-ruby1.9-2.0.5/lib/soap/mapping/encodedregistry.rb:216: warning: constant ::Fixnum is deprecated
DEPRECATION WARNING: `secrets.secret_token` is deprecated in favor of `secret_key_base` and will be removed in Rails 6.0. (called from <top (required)> at /usr/src/app/config/environment.rb:28)
Run options: exclude {:pact_live_events=>true}

Randomized with seed 28838
.........

Finished in 9.42 seconds (files took 27.48 seconds to load)
9 examples, 0 failures

Randomized with seed 28838